<launch>

    <!-- This node will launch MAVROS -->
    <include file="$(find mavros)/launch/apm.launch"/>
    <rosparam param="mavros/system_id">1</rosparam>
    <rosparam param="mavros/component_id">240</rosparam>
    <rosparam param="mavros/fcu_url">/dev/ttyACM0:115200</rosparam>
        
    <!-- 
    this argument determines the video address of camera (usually /dev/video0).
    You can check by writing v4l2-ctl list-devices in terminal 
    -->
    <arg name="camera_num" default="0" />
    
    <node name="vision_dropzone" pkg="rasendriya" type="vision_dropzone.py" args="$(arg camera_num)" output="screen"/>

    <param name="/rasendriya/dropping_altitude" value="15" />
    <param name="/rasendriya/focal_length/x" value="217.10032334" />
    <param name="/rasendriya/focal_length/y" value="232.0753471" />

    <param name="/rasendriya/wp_drop_first" value="10" />
    <param name="/rasendriya/wp_drop_second" value="18" />
    <param name="/rasendriya/wp_prepare_scan" value="3" />
    
    <node name="mission_control" pkg ="rasendriya" type="mission_control" output="screen"/>
    
</launch>
